spec/ics-003-connection-semantics/README.md

```diff
@@ -134,29 +134,86 @@ function removeConnectionFromClient(
 }
 ```
 
-Two helper functions are defined to provide automatic application of `CommitmentPrefix`. In the other parts of the specifications,
-these functions MUST be used for introspecting other chains' state, instead of directly calling the `verifyMembership` or `verifyNonMembership` function on the client.
+Helper functions are defined by the connection to pass the `CommitmentPrefix` associated with the connection to the verification function
+provided by the client. In the other parts of the specifications, these functions MUST be used for introspecting other chains' state,
+instead of directly calling the verification functions on the client.
 
 ```typescript
-function verifyMembership(
+function verifyClientConsensusState(
   connection: ConnectionEnd,
   height: uint64,
   proof: CommitmentProof,
-  path: Path,
-  value: Value): bool {
+  clientIdentifier: Identifier,
+  consensusState: ConsensusState) {
     client = queryClient(connection.clientIdentifier)
-    client.verifyMembership(height, proof, applyPrefix(connection.counterpartyPrefix, path), value)
+    return client.verifyClientConsensusState(connection, height, connection.counterpartyPrefix, proof, clientIdentifier, consensusState)
 }
-```
 
-```typescript
-function verifyNonMembership(
+function verifyConnectionState(
+  connection: ConnectionEnd,
+  height: uint64,
+  proof: CommitmentProof,
+  connectionIdentifier: Identifier,
+  connectionEnd: ConnectionEnd) {
+    client = queryClient(connection.clientIdentifier)
```

Comment:
should be client state

Reply:
Why? The idea is that the client exposes this function.